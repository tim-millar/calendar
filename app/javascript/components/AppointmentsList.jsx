import React from 'react';

import Appointment from './Appointment';

class AppointmentsList extends React.Component {
	render () {
		const appointments = this.props.appointments;

		return (
			appointments.map(function (appointment) {
				return (
					<Appointment
						key={appointment.id}
						title={appointment.title}
						time={appointment.time} />
				)
			})
		)
	}
}

export default AppointmentsList;
