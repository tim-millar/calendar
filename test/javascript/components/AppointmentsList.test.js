import React from 'react'
import { shallow } from 'enzyme'
import AppointmentsList from 'AppointmentsList'

describe('AppointmentsList', () => {
  it('should render a list of appointments', () => {
    const children = jest.fn()
    const appointments = []
    const appointmentsList = shallow(
      <AppointmentsList children={children} appointments={appointments} />,
    )
    expect(appointmentsList).toMatchSnapshot()
    expect(children).toHaveBeenCalledWith(appointments)
  })
})
