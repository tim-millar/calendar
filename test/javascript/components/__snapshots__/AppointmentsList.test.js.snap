// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentsList should render a list of appointments 1`] = `<div />`;
